### PR TITLE
KAFKA-8227 DOCS Fixed missing links duality of streams tables

### PR DESCRIPTION
**Pull request by londoncalling**
## Description

- Jira ticket: https://issues.apache.org/jira/browse/KAFKA-8227
- Related PR: https://github.com/apache/kafka-site/pull/204
- DOCS update to fix missing and broken links
 
## Review Requests
@bbejeck , @mjsax , @JimGalasyn , @joel-hamill 

Signed-off-by: Victoria Bialas <vicky@confluent.io>

### CODE REVIEW

File: docs/streams/core-concepts.html [hunk 1]
```diff
@@ -160,21 +160,26 @@ <h3><a id="streams_concepts_duality" href="#streams-concepts-duality">Duality of
...
         Any stream processing technology must therefore provide <strong>first-class support for streams and tables</strong>.
-        Kafka's Streams API provides such functionality through its core abstractions for
-        <code class="interpreted-text" data-role="ref">streams &lt;streams_concepts_kstream&gt;</code> and
-        <code class="interpreted-text" data-role="ref">tables &lt;streams_concepts_ktable&gt;</code>, which we will talk about in a minute.
-        Now, an interesting observation is that there is actually a <strong>close relationship between streams and tables</strong>,
-        the so-called stream-table duality.
-        And Kafka exploits this duality in many ways: for example, to make your applications
+        Kafka's Streams API provides such functionality through its core abstractions for 
+        <a id="streams_concepts_kstream" href="/{{version}}/documentation/streams/developer-guide/dsl-api#streams_concepts_kstream">streams</a>
+        <code class="interpreted-text" data-role="ref">&lt;streams_concepts_kstreams&gt;</code>
```
Comment (bbejeck) [inline docs/streams/core-concepts.html]: All the links point to the correct sections but I think we can get rid of all the `<code class="interpreted-text" data-role="ref">&lt;XX&gt;</code>` blocks in this section and elsewhere.  

I could be wrong, but It looked to me the `interpreted-text` sections were placeholders for setting the correct links pointing to other sections in the docs.

**Review by bbejeck — COMMENTED** [review]
@londoncalling thanks for the PR.  I just have some minor comments, but otherwise this PR LGTM.

**Comment by londoncalling** [comment]
@mjsax , @joel-hamill , @JimGalasyn this should be good to go after build completes, can someone merge for me who has permissions?

**Review by JimGalasyn — APPROVED** [review]

**Review by bbejeck — APPROVED** [review]
Thanks for the cleanup @londoncalling! LGTM

**Comment by bbejeck** [comment]
Merged #6625 into trunk

**Comment by bbejeck** [comment]
Thanks for fixing the links @londoncalling!

**Comment by bbejeck** [comment]
cherry-picked to 2.2, 2.1, and 2.0

**Pull request merged**